### ISSUE TITLE
Synthese: filtres par defaut

### ISSUE THREAD

**Issue by joelclems** [PnX-SI/GeoNature#2261]
Ce serait bien de pouvoir définir dans la configuration des filtres par défaut pour la page de visualisation de la synthèse.

Il pourrait s'agir par exemple de n'afficher que les observation de présence ou ne pas afficher les données invalidées.

**Comment by joelclems** [comment]
Déjà évoqué dans https://github.com/PnX-SI/GeoNature/issues/1377

**Comment by joelclems** [comment]
- [ ] Ajouter un paramètre `DEFAUT_FILTERS` à la configuration de la synthese
  - [ ] de type dictionnaire et qui contiendrait les valeurs par défaut du formulaire des filtres
    - par exemple `{id_nomenclature_observation_status: [87]}`
  - [ ] prise en compte dans le formulaire des filtre de la synthese
  - [ ] prise en compte dans le générateur de formulaire dynamique

**Comment by camillemonchicourt** [comment]
Oui, en effet ce sujet a été discuté dans plusieurs tickets.
La principale difficulté il me semble est que l'utilisateur capte que des filtres sont appliqués par défaut.

Quand des filtres par défaut sont définis, cela se répercute bien dans les filtres des formulaires ?
Concernant les statuts d'observation et de validation, ce sont des filtres avancés, donc tout en bas des filtres, donc pas très visible en l'état il me semble, mais certainement à laisser comme ça pour le moment.

Et il est possible de passer plusieurs id_nomenclatures pour un même filtre ? Par exemple plusieurs niveaux de validation ?

**Comment by joelclems** [comment]
J'ai testé pour `id_nomenclature_observation_status` et `id_dataset` seulement, et les valeurs s'affiche bien dans le formulaire (et le formulaire additionel pour `id_nomenclature_observation_status` s'affiche aussi.

On peut passer plusieurs nomenclature quand la valeur attendue est une liste d'id (comme pour les statut de validation)

On fournis ces valeurs au formulaire quasi telles quelles, (on s'assure juste que les champs additionels soient bien affichés)

Pour gagner du temps on peut regarder les paramètre de post de la requete appellée pour récupérer les données de la synthèse filtrées ( route `/synthese/for_web`)

**Comment by joelclems** [comment]
Je n'ai pas réussi à le faire marcher pour les cd_nom, sinon ça marche pour presque tout
Est ce que cela vaudrait le coup de pouvoir définir les filtres avec les cd_nomenclature  (à la place des id_nomenclature)? Ce serait un peu plus compliqué mais faisable et plus générique (un id_nomenclature peu varier d'une base à l'autre).

**Comment by camillemonchicourt** [comment]
C'est utile principalement (uniquement ?) pour les nomenclatures à mon avis.

Utiliser les id_nomenclature ou cd_nomenclature ? Les 2 me vont.

**Comment by camillemonchicourt** [comment]
Je pense que c'est OK comme ça avec les ID si c'est ce qu'utilise le frontend.

Dans la doc d'exemple, c'est intéressant si tu mets un autre exemple de nomenclature avec plusieurs ID (niveaux de validation je dirai).
Et c'est pas mal en effet de le mettre aussi dans le fichier exemple de configuration.

**Comment by DonovanMaillard** [comment]
C'est pas forcement choquant que la conf ne soit pas transposable d'une instance à l'autre, surtout si c'est documenté. D'autant qu'il semble préférable de ne pas mettre en place de filtres dans la configuration par défaut ? 

En tous cas, merci pour cette fonctionnalité qui va nous servir :)

**Comment by camillemonchicourt** [comment]
Un point de vigilance cependant : 
Je fais une recherche sur un taxon et j'ajoute une zone, je lance ma recherche. Les filtres avancés par défaut sont appliqués.

Je reinitialise mes filtres avec le bouton dédié à cela et je cherche autre chose dans une autre zone.
Les filtres avancés par défaut ne sont plus appliqués et je ne m'en rends pas forcément compte...

**Comment by joelclems** [comment]
Effectivement ce n'est pas simple et on peut des fois vouloir enlever tous les filtres et des fois garder ceux par défaut. 
Il faudrait presque avoir 2 bouttons de remise à zéros, un qui enlève tous les filtre et un qui remet les filtres par defaut, mais comment présenter ça pour que cela restes imple et intuitif ?

**Comment by camillemonchicourt** [comment]
Ouais non ça ferait trop de boutons.
Et avoir des filtres par défaut n'est pas la norme mais un cas d'usage.

On peut éventuellement imaginer que quand on reinitialise les filtres avec le bouton, il remette ceux par défaut.

**Comment by jupierrot** [comment]
Sera-t'il possible de bien ajouter dans la partie filtre les filtres actifs par défaut afin que l'utilisateur ai un visuel de ce sur quoi porte sa recherche ? Et qu'il puisse les supprimer si besoin !

**Comment by camillemonchicourt** [comment]
Oui si des filtres par défaut sont définis, il seront activés et visibles dans la panneau des filtres. C'est ce qui est discuté et indiqué plus haut.
Mais comme ce sont des filtres avancés, ce sera en bas du panneau, donc pas très très visible. Mais c'est déjà un bon début.

**Comment by jpm-cbna** [comment]
> Est ce que cela vaudrait le coup de pouvoir définir les filtres avec les cd_nomenclature (à la place des id_nomenclature)?

Il me semble que @bouttier a supprimé les références aux ID spécifiques à une installation de GeoNature dans le fichier de config. Du coup, je pense que malgré tout, il faudrait mieux utiliser les `cd_nomenclature`...

**Comment by camillemonchicourt** [comment]
Pas forcément, je vois pas pourquoi.
Vu que c'est un config spécifique à chaque instance de GeoNature, pas de soucis d'utiliser les ID.

**Comment by jpm-cbna** [comment]
> Pas forcément, je vois pas pourquoi.

Avec un id, il n'est pas possible de le partager ou de pouvoir définir une valeur par défaut car c'est dépendant de l'installation. Si le paramètre `DEFAULT_FILTERS` est un dictionnaire vide par défaut, c'est effectivement moins problématique.

**Comment by camillemonchicourt** [comment]
Oui c'est vide par défaut, et ça permet que ça soit que le frontend qui gère les filtres par défaut.

**Comment by bouttier** [comment]
Même si c’est spécifique à une instance donnée, l’utilisation de code reste préférable :
- C’est sémantiquement beaucoup plus lisible que des id dont on ne sait pas à quoi ils correspondent
- Cela permet de partager de la configuration entre plusieurs instances
- Il est possible de réinstaller une nouvelle base en gardant sa configuration
- Il est possible de déployer une configuration avec un outil de configuration automatique (puppet, ansible, …) sans devoir préalablement fouiller dans la base pour trouver les bons IDs

Joël tu disais que l’utilisation de code est plus compliqué que des IDs. J’imagine que c’est parce que l’API attends des IDs et non des codes. Note pour les futurs devs : je pense que l’utilisation de code plutôt que d’ID au niveau d’une API est de manière générale moins gênant que pour la configuration, mais reste préférable pour la sémantique, et pour pouvoir utiliser une API sans savoir ce qu’il y a dans la base qui est derrière.

**Comment by joelclems** [comment]
Effectivement, c'est plus clair et plus facile pour partager une configuration

Ce ne serait pas si compliqué à mettre en place, on pourrais charger l'ensemble des nomenclatures nécessaires et les utiliser ensuite.

**Comment by joelclems** [comment]
Il y a aussi un autre détail important, le composant des filtres `pnx-synthese-search` est aussi utilisé par le module validation.

Il faudrait faire en sorte de pouvoir préciser les valeurs par défaut comme input de ce composant pour pour spécifier des valeurs différentes pour la synthese et la validation

**Comment by camillemonchicourt** [comment]
Actuellement, pour ces filtres par défaut, si on veut utiliser des codes de nomenclatures plutôt que leurs identifiants, cela impliquerait quoi ?

Faire évoluer une API ?
Faire évoluer les composants frontend de sélection des nomenclatures ?

**Comment by joelclems** [comment]
Pour ne pas modifier l'api dans un premier temps. Cela impliquerai de charger les nomenclatures puis de faire la correspondance entre les codes et les id.

**Comment by joelclems** [comment]
utilisation des cd_nomenclature ici [5b2eb5c](https://github.com/PnX-SI/GeoNature/pull/2262/commits/5b2eb5cd4a2aa3c86df998833f6faa7a541fb12a)

ce serait peut être à mettre dans un nomenclature.service si ce genre de chose peut servir par ailleurs

**Comment by joelclems** [comment]
et aussi ajout d'un input à synthese-form pour ne pas réutiilser les même filtre dans la validation
